### PR TITLE
Add support for specifying country

### PR DESCRIPTION
**Pull request by n-thumann**
This PR adds support for specifying the country to search for or to download from.
Sometimes apps are not available in the US, which is the [default country](https://developer.apple.com/library/archive/documentation/AudioVideo/Conceptual/iTuneSearchAPI/Searching.html) and can therefore not be found.
An example for this issue is the German "Corona-Warn-App":
```
ipatool search "Corona-Warn–App" --log-level debug
==> 🛠	[Debug] Creating HTTP client...
==> 🛠	[Debug] Creating iTunes client...
==> ℹ️	[Info] Searching for 'Corona-Warn–App'...
==> ℹ️	[Info] Found 5 results:
1. SwissCovid: ch.admin.bag.dp3t (1.5.0).
2. HEALTHLYNKED COVID-19 Tracker: app.HealthLynked.COVAID19Tracker (1.0.64).
3. Commander Compass Go: com.paully.commandercompasslite (3.9.17).
4. iHandy Level: com.ihandysoft.carpenter.level (1.70.3).
5. Conceptis Fill-a-Pix: com.conceptispuzzles.Fill-a-Pix (5.6).
```
```
ipatool download -b de.rki.coronawarnapp --log-level debug
==> 🛠	[Debug] Creating HTTP client...
==> 🛠	[Debug] Creating iTunes client...
==> ℹ️	[Info] Querying the iTunes Store for 'de.rki.coronawarnapp'...
==> 🛠	[Debug] appNotFound
==> ❌	[Error] Could not find app.
```

With this patch applied:
```
ipatool search --country DE "Corona-Warn–App" --log-level debug
==> 🛠	[Debug] Creating HTTP client...
==> 🛠	[Debug] Creating iTunes client...
==> ℹ️	[Info] Searching for 'Corona-Warn–App' in country 'DE'...
==> ℹ️	[Info] Found 5 results:
1. Corona-Warn-App: de.rki.coronawarnapp (2.2.1).
2. HEALTHLYNKED COVID-19 Tracker: app.HealthLynked.COVAID19Tracker (1.0.64).
3. NINA: de.materna.bbk.mobile (3.3.3).
4. Corona-Datenspende: de.rki.coronadatenspende (2.2.1).
5. STIKO-App: de.rki.impf-infos (5.1).
```
```
ipatool download --country DE -b de.rki.coronawarnapp --log-level debug
==> 🛠	[Debug] Creating HTTP client...
==> 🛠	[Debug] Creating iTunes client...
==> ℹ️	[Info] Querying the iTunes Store for 'de.rki.coronawarnapp' in country 'DE'...
==> 🛠	[Debug] Found app: Corona-Warn-App (2.2.1).
==> ⚠️	[Warning] Enter Apple ID email:
[...]
```

### CODE REVIEW

File: Source/Commands/Search.swift [hunk 1]
```diff
@@ -16,6 +16,9 @@ struct Search: ParsableCommand {
     @Option
     private var limit: Int = 5
 
+    @Option(help: "The country to search in.")
```
Comment (majd) [inline Source/Commands/Search.swift:19]:
```suggestion
    @Option(help: "The two-letter (ISO 3166-1 alpha-2) country code for the store you want to search.")
```

File: Source/Commands/Search.swift [hunk 2]
```diff
@@ -26,16 +29,16 @@ struct Search: ParsableCommand {
...
         logger.log("Creating HTTP client...", level: .debug)
         let httpClient = HTTPClient(urlSession: URLSession.shared)
 
         logger.log("Creating iTunes client...", level: .debug)
         let itunesClient = iTunesClient(httpClient: httpClient)
         
         do {
-            logger.log("Searching for '\(term)'...", level: .info)
-            let results = try itunesClient.search(term: term, limit: limit)
+            logger.log("Searching for '\(term)' in country '\(country)'...", level: .info)
```
Comment (majd) [inline Source/Commands/Search.swift:40]:
```suggestion
            logger.log("Searching for '\(term)' using the '\(country)' store front...", level: .info)
```

**Review by majd — CHANGES_REQUESTED** [review]

**Comment by majd** [comment]
Thanks for contributing! Happy to get this merged once the requested changes are made. 😄

**Comment by n-thumann** [comment]
Applied both of your suggestions and squashed them ✌🏻

**Review by majd — APPROVED** [review]

**Pull request merged**